### PR TITLE
Kf 4432 fix cherry pick images config

### PR DESCRIPTION
**Pull request by i-chvets**
# Description

Cherry-picking commits to ensure images are configurable for airgapped environment.

More details are in https://github.com/canonical/istio-operators/issues/332

Only the following commits were merged to support image configuration in airgapped environment:
- 3974636
- 293f90a
- bed0151

### CODE REVIEW

**Review by orfeas-k — DISMISSED** [review]
Approving but CI should pass first. Could this be related to `track/1.16` missing this PR https://github.com/canonical/istio-operators/pull/326? The errors don't match TBH but I noticed [the tests are deployed](https://github.com/canonical/istio-operators/actions/runs/6318914992/job/17162382286?pr=334#step:6:441) in `cos-test` model

**Review by DnPlas — COMMENTED** [review]
Hey @i-chvets, could you please rebase your branch with git to get rid of the two merge commits that were introduced:
* 17ba45f01e641354fe9fafea5498442730bcbf72
* 157bb6cc1de93cb6b2e52b3b034f574ef74309cb

That way we keep the commit list as clean as possible.

File: charms/istio-pilot/src/charm.py [hunk 1]
```diff
@@ -156,19 +158,42 @@ def __init__(self, *args):
...
                 "./istioctl",
                 "install",
                 "-y",
-                "-s",
+                "--set",
                 "profile=minimal",
-                "-s",
-                f"values.global.istioNamespace={self.model.name}",
+                "--set",
+                "values.global.istioNamespace=kubeflow",
```
[the remark this answers is not in the document]
Comment (i-chvets) [reply charms/istio-pilot/src/charm.py]: Done. Should be in this PR now. Let's wait for CI to finish.

**Comment by i-chvets** [comment]
Cleaned up merge commits.

**Review by DnPlas — APPROVED** [review]

**Pull request merged**